src/lib.rs

```diff
@@ -235,24 +284,28 @@ fn model(state: &Mutex<AppState>, audio: AudioManager) {
 
 /// Validates a config file, or exits the program with an error printed to the command line on how
 /// to fix the validation
-fn validate_config(config: &mut config::Config) {
+fn validate_config(config: &mut config::Config) -> Result<()> {
     let alarm_names: Vec<_> = config.alarms.iter().map(|b| &b.name).collect();
+    // Loops through each dial and checks if its corresponding alarm exists in the map
     for row in &config.dial_rows {
         for dial in &row.dials {
             let alarm_name = &dial.alarm;
             if !alarm_names.contains(&alarm_name) {
-                eprintln!("Alarm `{alarm_name}` is missing!");
-                eprintln!("Available alarms are {alarm_names:?}");
-
                 let message = format!(
                     "Alarm `{alarm_name}` is missing!\nAvailable alarms are: {alarm_names:?}"
                 );
-                let popup =
-                    ErrorPopup::new("Configuration Error", "Invalid configuration", message);
+
+                let popup = DialogPopup::new(
```

Comment:
Looks like `eframe::run_native` just got updated to not return `!` in this latest version you updated to: https://github.com/emilk/egui/pull/1889

Before `popup.show()` never returned because of this, so I made this a function that never returns in my pr: https://github.com/TheTrioWLT/dials-research/pull/35/files#diff-99682eb97a9d1693dbb3423ce95a6df9c6f57edac19f97fdc85da7353084099dR4

But this seems a lot better to give the user the opportunity to click ok and then have bail give the error to main so it will panic and show if they are running in the terminal. Ill update my pr after we merge this one to account for this. Great stuff!

Reply:
Cool

Reply:
Yeah I saw that, sounds good!